packages/node_modules/@webex/webex-core/src/lib/services/service-catalog.js

```diff
@@ -106,6 +108,110 @@ const ServiceCatalog = AmpState.extend({
     return this;
   },
 
+  /**
+   * Search over all service groups to find a cluster id based
+   * on a given url.
+   * @param {string} url - Must be parsable by `Url`
+   * @returns {string} - ClusterId of a given url
+   */
+  findClusterId(url) {
+    const incomingUrlObj = Url.parse(url);
+    let clusterId;
+    let serviceUrlObj;
+
+    Object.keys(this.serviceGroups).forEach((key) => {
+      this.serviceGroups[key].forEach((service) => {
+        serviceUrlObj = Url.parse(service.defaultUrl);
+
+        if (serviceUrlObj.host === incomingUrlObj.host &&
+          service.hosts.length > 0) {
+          clusterId = service.hosts[0].id;
```

Comment:
Should we set this up so that it breaks out of the loops once it finds an ID?

Reply:
There isn't a built-in break method for `Array.forEach()`. This loop would need to be converted to a `for(iterator, conditional, increment)` loop. Let me know if you think this is necessary.

Reply:
Yea, let's change it to the `for` iterator so we can break out of the loops.

Reply:
addressed